src/OpenLoco/src/World/CompanyAi.cpp

```diff
@@ -2904,23 +3019,236 @@ namespace OpenLoco
+        {
+            return GameCommands::FAILURE;
+        }
+        uint8_t _112C2F4 = 0;
+        uint8_t roadStationObjId = 0xFFU;
+        if (!elRoad->isAiAllocated() && elStation != nullptr)
+        {
+            _112C2F4 |= 1U << 0; // hasNonAiAllocatedRoad
+        }
+        if (elStation != nullptr)
+        {
+            _112C2F4 |= 1U << 1; // hasStation
+            roadStationObjId = elStation->objectId();
+            if (noStations)
+            {
+                return GameCommands::FAILURE;
+            }
+        }
+
+        currency32_t totalCost = 0;
+        const auto existingRoadObjId = elRoad->roadObjectId();
+        auto* existingRoadObj = ObjectManager::get<RoadObject>(existingRoadObjId);
+        uint8_t mods = 0;
+        if (!existingRoadObj->hasFlags(RoadObjectFlags::unk_03))
+        {
+            mods = elRoad->mods();
+        }
+        uint8_t bridge = 0xFFU;
+        if (elRoad->hasBridge())
+        {
+            bridge = elRoad->bridge();
+        }
+
+        if (elStation != nullptr)
+        {
+            GameCommands::RoadStationRemovalArgs args{};
+            args.pos = pos;
+            args.roadId = roadId;
+            args.roadObjectId = existingRoadObjId;
+            args.rotation = rotation;
+            args.index = sequenceIndex;
+            GameCommands::doCommand(args, GameCommands::Flags::apply | GameCommands::Flags::aiAllocated | GameCommands::Flags::noPayment);
+        }
+
+        if (!(_112C2F4 & (1U << 0)))
+        {
+            assert(existingRoadObjId == roadObjectId);
+            GameCommands::RoadRemovalArgs args{};
+            args.pos = pos;
+            args.objectId = existingRoadObjId;
+            args.roadId = roadId;
+            args.rotation = rotation;
+            args.sequenceIndex = sequenceIndex;
+            GameCommands::doCommand(args, GameCommands::Flags::apply | GameCommands::Flags::aiAllocated | GameCommands::Flags::noPayment);
+            // Vanilla copied the bridge here via a global but
+            // we have already previously done that
+        }
+
+        if (!(_112C2F4 & (1U << 0)))
+        {
+            GameCommands::RoadPlacementArgs args{};
+            args.bridge = bridge;
+            args.mods = mods;
+            args.pos = pos;
+            // We shift this down as we need to be at the start of the road piece
+            // which for road that is sloping will not be the same as the
+            // produced road element
+            args.pos.z -= TrackData::getRoadPiece(roadId)[0].z;
+            args.roadId = roadId;
+            args.roadObjectId = roadObjectId;
+            args.rotation = rotation;
+            auto res = GameCommands::doCommand(args, GameCommands::Flags::apply | GameCommands::Flags::flag_7);
+            if (res == GameCommands::FAILURE)
+            {
+                res = GameCommands::doCommand(args, GameCommands::Flags::apply | GameCommands::Flags::noPayment | GameCommands::Flags::flag_7);
```

Comment:
"If we can't pay for it, we'll just get it for free"?? 🥲

Reply:
Ai are a bunch of cheats